CIP-0067/README.md

```diff
@@ -0,0 +1,93 @@
+---
+CIP: 67
+Title: Asset Name Label Registry
+Authors: Alessandro Konrad <alessandro.konrad@live.de>, Thomas Vellekoop <thomas.vellekoop@iohk.io>
+Status: Draft
+Type: Informational
+Created: 2022-07-13
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This proposal defines a standard to classify Cardano native assets by the asset name.
+
+## Motivation
+
+As more assets are minted and different standards emerge to query data for these assets, it's getting harder for 3rd parties to determine the asset type and how to proceed with it. This standard is similar to [CIP-0010](../CIP-0010), but focuses on the asset name of an asset.
+
+
+## Specification
+
+To classify assets the `asset_name` needs to be prefixed the following `4 bytes` binary encoding:
+```
+[ 0000 | 2 bytes label_num | 1 byte checksum | 0000 ]
+```
+- The leading and ending four 0s are brackets
+- `label_num` has a fixed size of 2 bytes (`Label range in decimal: [0, 65535]`). 
+If `label_num` < 2 bytes the remaining bits need to be padded with 0s.
+- `checksum` has a fixed size of 1 byte. The checksum is calculated by applying the CRC-8 algorithm on the `label_num (including the padded 0s)`. 
+ 
```

Comment:
> the CRC-8 algorithm

There is no one canonical CRC 8 algorithm, each one uses a specific irreducible polynomial over GF(2) (a finite field over a binary field). We should specify which irreducible polynomial we take for the standard. Preferably, we denote this polynomial in its binary form, so as an example
```
x^8 + x^4 + x^3 + x^2 + 1 = 1*x^8 + 0*x^7 + 0*x^6 + 0*x^5 + 1*x^4 + 1*x^3 + 1*x^2 + 0*x^1 + 1*x^0
```
becomes `100011101` in its binary representation. I found online the following primitive polynomials of degree 8 for GF(2)
```
x^8 + x^4 + x^3 + x^2 + 1
x^8 + x^5 + x^3 + x^1 + 1
x^8 + x^6 + x^4 + x^3 + x^2 + x^1 + 1
x^8 + x^6 + x^5 + x^1 + 1
x^8 + x^6 + x^5 + x^2 + 1
x^8 + x^6 + x^5 + x^3 + 1
x^8 + x^7 + x^6 + x^1 + 1
x^8 + x^7 + x^6 + x^5 + x^2 + x^1 + 1
```

Reply:
What about providing the lookup table also?